### ISSUE TITLE
java-hamcrest vs. hamcrest-core, hamcrest-library

### ISSUE THREAD

**Issue by Jeff-Walker** [hamcrest/JavaHamcrest#183]
Maybe I'm missing something, but I'm looking for the latest version of hamcrest. In the past I've always used hamcrest-core, etc. and the latest version is 1.3 from 2012. Now it seems there is java-hamcrest version 2.0 that's newer. Spring boot starter has hamcrest 1.3, but java-hamcrest has things I want. I can exclude/include the libraries in my POM, but it's very confusing.

I don't see anything in the docs or the GitHub project pages that indicates the name change. Moreover, when I click on the Maven central link it takes me to the search for "org.hamcrest" which includes both.

After some digging I've decided that I want java-hamcrest, but as a longtime users, it took me a while to figure this out.

**Comment by StevenGBrown** [comment]
Do I want hamcrest-junit or java-hamcrest? Maybe both? I agree that some documentation would be very useful. :+1:

**Comment by sf105** [comment]
good point. It was a work in progress.

**Comment by Thunderforge** [comment]
@sf105 So what will actually be done to resolve this? Do we change documentation? Do we add a release to [the GitHub releases tab](https://github.com/hamcrest/JavaHamcrest/releases)? How will we resolve this issue of people not knowing what to download?

**Comment by garretwilson** [comment]
Yes, can someone explain this? All our code is using `hamcrest-library`, and I'm teaching that in the course I've written and given for a few years. Now on [Baeldung: Hamcrest Text Matchers](https://www.baeldung.com/hamcrest-text-matchers) I see a reference to `java-hamcrest`, and I have no idea what it is. Does it replace `hamcrest-library`? Does it have everything that `hamcrest-library` does? Is it newer/better? Specifically I'm wanting the `matchesPattern(…)` Baeldung mentions. Is this a new addition?

At the least I just need something telling me what `java-hamcrest` is and its relation to `hamcrest-library`. Where can I find that?

Thank you kindly in advance.

**Comment by garretwilson** [comment]
P.S. If the "official page" that tells what's going on is continually delayed until "tomorrow", could someone at least provide an informal answer here in this ticket? That would be better than nothing. After all, this ticket is already coming up in searches for people (like me) wondering what the story is.

**Comment by sf105** [comment]
`java-hamcrest` was a failed attempt at a new packaging. I think the best answer would be to look at `hamcrest-all` for future reference. @tumbarumba should be able to provide a more up-to-date answer.
